packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -551,32 +562,38 @@ export class CommonFrontendContribution implements FrontendApplicationContributi
             execute: (event?: Event) => {
                 const tabBar = this.findTabBar(event)!;
                 const currentIndex = tabBar.currentIndex;
-                this.shell.closeTabs(tabBar, (_, index) => index > currentIndex);
+                this.shell.closeTabs(tabBar, (title, index) => index > currentIndex && title.closable);
             }
         });
         commandRegistry.registerCommand(CommonCommands.CLOSE_ALL_TABS, {
-            isEnabled: (event?: Event) => this.findTabBar(event) !== undefined,
-            execute: (event?: Event) => this.shell.closeTabs(this.findTabArea(event)!)
+            isEnabled: (event?: Event) => {
+                const tabBar = this.findTabBar(event);
+                return tabBar !== undefined && tabBar.titles.some(title => title.closable);
```

Comment:
This condition checks only the selected tab bar, while `execute` closes all other tab bars, too.

Reply:
Same thing here seems the original logic was wrong.